### PR TITLE
target_dependency: Add target proxy to `to_tree_hash`.

### PR DESCRIPTION
**Pull request by byohay**
The added information helps identifying the correct dependency.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* target_dependency: Add target proxy to `to_tree_hash`.  
  [byohay](https://github.com/byohay)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
@byohay is there a test you can add for this?

**Comment by byohay** [comment]
@dnkoutso done.

**Comment by byohay** [comment]
@dnkoutso anything else to add?

File: spec/project/object/target_dependency_spec.rb [hunk 1]
```diff
@@ -40,6 +40,32 @@ module ProjectSpecs
       lambda { @target_dependency.sort_recursively }.should.not.raise
     end
 
+    it 'tree hash does not contain target' do
```
Comment (amorde) [inline spec/project/object/target_dependency_spec.rb:43]: nit: could this specify that it includes the `targetProxy`?

Comment (byohay) [reply spec/project/object/target_dependency_spec.rb:43]: I thought about it, but then I'd have to include the `display_name ` and `isa` too, which makes the test name too verbose IMO :)

**Comment by byohay** [comment]
@amorde can we merge this?

**Pull request merged**